src/main.rs

```diff
@@ -144,31 +152,44 @@ fn try_main() -> Result<()> {
     Ok(())
 }
 
+#[derive(Tabled)]
+struct PackageStatus<'a> {
+    #[tabled(rename = "Name")]
+    name: &'a String,
+    #[tabled(rename = "Status")]
+    status: String,
+}
+
 /// Displays whether each package needs an update or not.
-fn log_summary(
-    pkgs: &BTreeMap<String, Package>,
-    new_vers: &BTreeMap<String, Version>,
-    old_vers: &BTreeMap<String, Version>,
-) {
-    if let Some(max_len) = pkgs.keys().map(String::len).max() {
-        info!("Results:");
-
-        for pkg in pkgs.keys() {
-            let new_ver = new_vers.get(pkg).unwrap();
-            info!(
-                "    {:<max_len$}  {}",
-                pkg,
-                old_vers.get(pkg).map_or_else(
+fn log_summary(new_vers: &BTreeMap<String, Version>, old_vers: &BTreeMap<String, Version>) {
+    if new_vers.is_empty() {
+        info!("No Packages installed");
+        return;
+    }
+
+    let pkgs: Vec<PackageStatus> = new_vers
+        .iter()
+        .map(|(name, new_ver)| {
+            (
+                name,
+                old_vers.get(name).map_or_else(
                     || format!("ø -> {new_ver}"),
-                    |old_ver| if old_ver < new_ver {
-                        format!("{old_ver} -> {new_ver}")
-                    } else {
-                        "✔".to_owned()
-                    }
+                    |old_ver| {
+                        if old_ver < new_ver {
+                            format!("{old_ver} -> {new_ver}")
+                        } else {
+                            format!("✔ {new_ver}")
+                        }
+                    },
                 ),
-            );
-        }
-    }
+            )
+        })
+        .map(|(name, status)| PackageStatus { name, status })
+        .collect();
+
+    let table = tabled::Table::new(pkgs).with(Style::sharp()).to_string();
+
+    info!("\n{table}");
```

Comment:
```suggestion
    info!("Results:\n{table}");
```

maybe? It would keep the existing look and avoid a weird log line that just starts with a newline.

Reply:
done